test/test_variable_ops.cpp

```diff
@@ -502,6 +502,273 @@ TEST(VariableTest, SmallGraph6) {
+  auto init_value = ops::Const(root, {{1.f, 1.f}, {1.f, 1.f}});
+  auto var_assign = ops::Assign(root.WithOpName("Assign1"), var, init_value);
+
+  auto accum = ops::Variable(root.WithOpName("accum"), varShape, DT_FLOAT);
+  auto init_value2 = ops::Const(root, {{3.f, 3.f}, {3.f, 3.f}});
+  auto accum_assign =
+      ops::Assign(root.WithOpName("Assign2"), accum, init_value2);
+
+  auto grad = ops::Const(root, {{2.f, 2.f}, {2.f, 2.f}});
+
+  auto lr = ops::Const(root, 1.f);
+
+  ops::ApplyAdagrad::Attrs use_locking;
+  use_locking = use_locking.UseLocking(true);
+  auto applyadagrad_t = ops::ApplyAdagrad(root.WithOpName("Adagrad"), var,
+                                          accum, lr, grad, use_locking);
+
+  auto var1 = ops::Variable(root.WithOpName("Var1"), varShape, DT_FLOAT);
+  auto init_value1 = ops::Const(root, {{2.f, 3.f}, {4.f, 5.f}});
+  auto var1_assign = ops::Assign(root, var1, init_value1);
+
+  auto assign_add =
+      ops::AssignAdd(root.WithOpName("AssignAdd"), var1, applyadagrad_t);
+
+  // Turn off optimizations so that all the nodes are processed
+  tensorflow::SessionOptions options;
+  options.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(tensorflow::OptimizerOptions_Level_L0);
+  options.config.mutable_graph_options()
+      ->mutable_rewrite_options()
+      ->set_constant_folding(tensorflow::RewriterConfig::OFF);
+
+  // Run on nGraph
+  ActivateNGraph();
+  ClientSession ng_session(root, options);
+  std::vector<tensorflow::Tensor> ng_outputs1;
+  std::vector<tensorflow::Tensor> ng_outputs2;
+  std::vector<tensorflow::Tensor> ng_outputs3;
+  std::vector<tensorflow::Tensor> ng_outputs4;
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(ng_session.Run({{var_assign, accum_assign, var1_assign}},
+                             &ng_outputs1));
+  }
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(ng_session.Run({applyadagrad_t}, &ng_outputs2));
+  }
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(ng_session.Run({assign_add}, &ng_outputs3));
+  }
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(ng_session.Run({var1}, &ng_outputs4));
+  }
+
+  DeactivateNGraph();
+
+  // Run on TF
+  ClientSession tf_session(root, options);
+  std::vector<tensorflow::Tensor> tf_outputs1;
+  std::vector<tensorflow::Tensor> tf_outputs2;
+  std::vector<tensorflow::Tensor> tf_outputs3;
+  std::vector<tensorflow::Tensor> tf_outputs4;
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(tf_session.Run({{var_assign, accum_assign, var1_assign}},
+                             &tf_outputs1));
+  }
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(tf_session.Run({applyadagrad_t}, &tf_outputs2));
+  }
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(tf_session.Run({assign_add}, &tf_outputs3));
+  }
+
+  for (int i = 0; i < 10; i++) {
+    ASSERT_OK(tf_session.Run({var1}, &tf_outputs4));
+  }
+
+  Compare(tf_outputs1, ng_outputs1);
+  Compare(tf_outputs2, ng_outputs2);
```

Comment:
We don't need the string comparison coz of the sync optimizers. is that correct?

Reply:
This would need string comparison too for the initialization. I will update it now that the other PR that has that change is merged.